Dockerfile

```diff
@@ -1,27 +1,8 @@
-# Build the manager binary
-FROM golang:1.15 as builder
+FROM alpine:3.13
```

Comment:
So here we are dropping multistage docker build and using the local build to get the binary, right? My concern with this is that we become very dependent on the environment of who/what is building the project. This will use golang version of the machine, for example, instead of golang version specifically chosen by us. It is very clean though :thinking: 

Another thing. Wouldn't it be better to still use a distroless image instead of alpine?

Reply:
I think eventually distroless will be preferred, but I would like[ ephemeral containers](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/) to be more mature and available, otherwise debugging can be difficult. Alpine is pretty light weight.

In terms of builds, its a good point, but should still work on other systems if GOARCH and GOOS is specified. For published images, they are all built on the github runners running ubuntu-18 currently.

Reply:
+1 for more debuggable image while we don't have more mature ephemeral containers feature on k8s.

Another problem with this is that our Dockerfile is now also somewhat dependent on our Makefile (+ local environment). A standalone Dockerfile seems more clear to externals I think.

Reply:
One reason I prefer to remove the multi-stage docker build was to take advantage of caching modules and build stages for lint, testing, and building in CI.

I think depending on Makefile for build tasks is fine, it can ensure builds are created with a known and common set of flags/configuration. What I can do is split `build` into two separate targets, one which forces OS/ARCH for linux/amd64 and is used by the docker build target, and another that builds for local environment os/arch for use running locally.
Let me know if that would help with your concerns here.

Reply:
Taking advantage of caching modules makes a lot of sense.

Gonna approve the PR, consider splitting targets a Nit.

Keeping the Dockerfile standalone is a personal preference, however, looking and the thread, I am fine with the reasons to remove it :+1: 

Also, if crossplane does it, should be fine :smile: